### PR TITLE
fix bug traverseByLabel() not set page every loop

### PR DESCRIPTION
**Pull request by zhoney**
fixed: #802 #814

Change-Id: I586032e221ebfdc3141fb9374bca23a4d5af229c

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 1]
```diff
@@ -127,6 +127,7 @@ public GraphTransaction(HugeGraph graph, BackendStore store) {
                                 CoreOptions.VERTEX_CHECK_CUSTOMIZED_ID_EXIST);
         this.verticesCapacity = conf.get(CoreOptions.VERTEX_TX_CAPACITY);
         this.edgesCapacity = conf.get(CoreOptions.EDGE_TX_CAPACITY);
+        this.traverseBatch = conf.get(CoreOptions.QUERY_TRAVERSE_BATCH);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java]: prefer to use page_size

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/EdgeLabelCoreTest.java [hunk 2]
```diff
@@ -873,6 +875,43 @@ public void testRemoveEdgeLabelWithEdgeAndSecondaryIndex() {
         });
     }
 
+    @Test
+    public void testRebuildIndexOfEdgeLabelWithoutLabelIndex() {
+        Assume.assumeFalse("Support query by label",
+                           storeFeatures().supportsQueryByLabel());
+
+        initDataWithoutLabelIndex();
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/EdgeLabelCoreTest.java]: add case graph().traversal().E().hasLabel(xx)

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java [hunk 3]
```diff
@@ -1401,6 +1401,77 @@ public void testRebuildIndexLabelOfEdge() {
         Assert.assertNotNull(edge);
     }
 
+    @Test
+    public void testRebuildIndexOfVertexWithoutLabelIndex() {
+        Assume.assumeFalse("Support query by label",
+                           storeFeatures().supportsQueryByLabel());
+
+        initDataWithoutLabelIndex();
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java]: add case graph().traversal().V().hasLabel(xx)

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 4]
```diff
@@ -1538,10 +1539,14 @@ public void traverseEdgesByLabel(EdgeLabel label, Consumer<Edge> consumer,
                     SchemaLabel elemLabel = ((HugeElement) e).schemaLabel();
                     if (label.equals(elemLabel)) {
                         consumer.accept(e);
+                        if (deleting) {
+                            this.commit();
+                        }
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java]: call commitIf()

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 5]
```diff
@@ -1538,10 +1539,14 @@ public void traverseEdgesByLabel(EdgeLabel label, Consumer<Edge> consumer,
...
                     }
                 }
                 if (query.paging()) {
                     page = PageInfo.pageState(iter).toString();
+                    query.page(page);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java]: this.commit();

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 6]
```diff
@@ -127,6 +127,7 @@ public GraphTransaction(HugeGraph graph, BackendStore store) {
                                 CoreOptions.VERTEX_CHECK_CUSTOMIZED_ID_EXIST);
         this.verticesCapacity = conf.get(CoreOptions.VERTEX_TX_CAPACITY);
         this.edgesCapacity = conf.get(CoreOptions.EDGE_TX_CAPACITY);
+        this.traverseBatch = conf.get(CoreOptions.QUERY_PAGE_SIZE);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java]: rename this.traverseBatch to pageSize

**Comment by zhoney** [comment]
Also fixed #814

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/EdgeLabelCoreTest.java [hunk 7]
```diff
@@ -873,6 +875,65 @@ public void testRemoveEdgeLabelWithEdgeAndSecondaryIndex() {
...
+
+        // Not support query by label
+        Assert.assertThrows(NoIndexException.class, () -> {
+            graph().traversal().E().hasLabel("read").toList();
+        }, e -> {
+            Assert.assertTrue(
+                   e.getMessage().startsWith("Don't accept query by label") &&
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/EdgeLabelCoreTest.java]: use assertContains() instead and rebase on https://github.com/hugegraph/hugegraph-common/pull/43

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/EdgeLabelCoreTest.java [hunk 8]
```diff
@@ -873,6 +875,65 @@ public void testRemoveEdgeLabelWithEdgeAndSecondaryIndex() {
...
+                   e.getMessage().endsWith("it disables label index"));
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/EdgeLabelCoreTest.java]: use assertContains() instead and rebase on https://github.com/hugegraph/hugegraph-common/pull/43

**Pull request merged**